### ISSUE TITLE
[Network] opacity option for nodes

### ISSUE THREAD

**Issue by rwu823** [almende/vis#1185]
The edges have `color.opacity` options, how is node?

**Comment by AlexDM0** [comment]
Hi,

you could set the colors of the nodes by using the colors with rgba. There's no opacity there at the moment. I'm not sure when this will be added though. It would be nice to have them consistent.

Regards

**Comment by rwu823** [comment]
Yeah rgba fine, but I'm use `shape` :).  hope it would be enhancement on node

**Comment by AlexDM0** [comment]
shape?

**Comment by rwu823** [comment]
```js
{
  shape: 'image',
}
```

**Comment by AlexDM0** [comment]
I see. It would be nice if that was possible, I agree. Maybe in future releases :).

Regards

**Comment by mmcnl** [comment]
Being able to update the opacity of a node independently from the rest of its existing color options would be very helpful for my application. Being able to do something like this
```
nodes.update([{"id":1998, "color":{"opacity":0.3}}]);
``` 
would mean that my updating code could change the opacity without having to maintain the state of the existing color of the node (when changing opacity via `rgba`).

**Comment by pradeepacc** [comment]
Hi. Is this enhancement available now?

**Comment by wimrijnders** [comment]
Hi there! This issue has been labeled `Inactive`, because there has been no activity here for over two months. If the inactivity continues, the issue will be closed in two weeks.

If you feel that the issue should remain open, please comment here, preferably with motivation. Thanks!

**Comment by mmcnl** [comment]
Hi @wimrijnders, I'm still interested in this issue. I mention my motivation in https://github.com/almende/vis/issues/1185#issuecomment-277765175. Thanks!

**Comment by wimrijnders** [comment]
OK noted.

**Comment by wimrijnders** [comment]
I've been looking into this, it's actually a pretty tough one to get right.

It's relatively easy to add `opacity` to the node color; however, what would this mean for the contents of a node? E.g. a node can contain text and/or graphics. Should these also be affected by the opacity?

If so, then `opacity` is not a color option but a node option. Strictly speaking, in the example given:
```
nodes.update([{"id":1998, "color":{"opacity":0.3}}]);
```
...only the opacity of the color would be affected, the rest of the screen elements wouldn't.

-----

I would like some feedback on this from the commenters in this issue. The question is:

> Do you expect the option `opacity` to change the opacity of *all* node contents?

Because, if so, I propose to make it a node option, not an option of the color of the node.

**Comment by mmcnl** [comment]
Thanks for looking into this @wimrijnders. For my use case, being able to update the opacity of the entire node and its contents would be great.

**Comment by mmcnl** [comment]
However, a bit off topic, I've just realized that I need collapsible nodes (and I'm using hierarchical layout), so unfortunately it appears I'm going to have to look for another js library.

**Comment by wimrijnders** [comment]
I'm OK with that. But would you mind explaining what you mean by 'collapsible nodes?' Do you perhaps mean combining related nodes to a single one, as in example `network/other/clustering`?

**Comment by mmcnl** [comment]
Sure, I'd like to be able to click on a node and have it collapse all of its children/subchildren, eg like this: https://bl.ocks.org/d3noob/43a860bc0024792f8803bba8ca0d5ecd

I looked into the visjs clustering but saw some issues where the developers stated that it doesn't work with hierarchical layout.

**Comment by wimrijnders** [comment]
> ... the developers stated that it doesn't work with hierarchical layout.

Challenge accepted. Will halt all current activity and look into this *now*.

**Comment by wimrijnders** [comment]
@mmcnl Clustering works with hierarchical, subject to certain constraints. I would like to know your use case. Would you mind opening an issue for this. so we can discuss it without adding off-topic cruft to the current issue? That is, if you're willing to pursue the possibility. Thanks!

**Comment by mmcnl** [comment]
Great, thanks again for looking into it. Are there some docs or examples (or code) that would help me assess if this will work for my use case? I will try to write something up soon.

**Comment by wimrijnders** [comment]
@mmcnl Just open an issue with minimal commenting (e.g. get the title right) now and we'll take it from there.

**Comment by jamesbebbington** [comment]
Hey @wimrijnders our app would also greatly benefit from being able to set the opacity of entire nodes. Is this something you're currently working on? Thanks.

**Comment by kgiannelos** [comment]
+1 for adding an opacity option to the nodes module.
We are using vis.js for our application as well and we just came up with the need to set the opacity of the node and all of it's contents.

**Comment by cacois** [comment]
> Do you expect the option opacity to change the opacity of all node contents?
Yes, I would expect it to change the entire contents of the node. This seems the most logical behaviors for most applications I can think of.
